### PR TITLE
AMP-95395 - add marketing attribution tracking

### PR DESCRIPTION
**Pull request by yuhao900914**
# Amplitude Developer Docs PR

## Description

Describe your changes. 

## Deadline

When do these changes need to be live on the site?

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [ x ] Doc update.
- [ x ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [ ] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [ ] Running `mkdocs serve` didn't generate any failures.
- [ ] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/data/sdks/browser-2/index.md [hunk 1]
```diff
@@ -437,4 +437,25 @@ To disable offline mode, add `offline: amplitude.Types.OfflineDisabled` to the `
...
+
+We automatically track marketing attribution by default. Once marketing attribution tracking is enabled, Amplitude will automatically generate identify events to assign the campaign value in certain cases. This ensures that user properties are updated and will influence future events. Below, we outline scenarios that demonstrate when marketing attribution will or will not be tracked. It's important to remember that these examples are illustrative and not exhaustive:
+
+Tracking occurs when either of the following applies:
+
+|Rule|Example|
+|-|-|
+| A specific referrer domain is explicitly excluded.| When config the `config.defaultTracking.attribution.excludeReferrers` to include `[a.test.com]`, and the page has `a.test.com` as a referrer domain. |
+| No previous campaign. | A user's initial visit. |
+| There is an introduction with new UTM parameter or clickId parameter. | If any utm parameters or click id parameters has been droped during a session, we will unset it.|
```
Comment (justin-fiedler) [inline docs/data/sdks/browser-2/index.md:452]:
```suggestion
| There is an introduction of new UTM parameter or `clickId` parameter. | If any utm parameters or click id parameters has been dropped during a session, we will unset it. |
```

**Review by justin-fiedler — APPROVED** [review]

File: docs/data/sdks/browser-2/index.md [hunk 2]
```diff
@@ -437,4 +437,25 @@ To disable offline mode, add `offline: amplitude.Types.OfflineDisabled` to the `
 amplitude.init(AMPLITUDE_API_KEY, {
   offline: amplitude.Types.OfflineDisabled
 });
-```
\ No newline at end of file
+```
+
+### Marketing Attribution Tracking
+
+We automatically track marketing attribution by default. Once marketing attribution tracking is enabled, Amplitude will automatically generate identify events to assign the campaign value in certain cases. This ensures that user properties are updated and will influence future events. Below, we outline scenarios that demonstrate when marketing attribution will or will not be tracked. It's important to remember that these examples are illustrative and not exhaustive:
```
Comment (markzegarelli) [inline docs/data/sdks/browser-2/index.md:444]:
```suggestion
Amplitude tracks marketing attribution by default. Once you enable marketing attribution tracking, Amplitude generates `identify` events to assign the campaign value in certain cases. This ensures that user properties update and influence future events. 

For more information, see the scenarios outlined below that demonstrate when Amplitude does or doesn't track marketing attribution. These examples are illustrative, not exhaustive:
```

File: docs/data/sdks/browser-2/index.md [hunk 3]
```diff
@@ -437,4 +437,25 @@ To disable offline mode, add `offline: amplitude.Types.OfflineDisabled` to the `
...
+Tracking occurs when either of the following applies:
+
+|Rule|Example|
+|-|-|
+| A specific referrer domain is explicitly excluded.| When config the `config.defaultTracking.attribution.excludeReferrers` to include `[a.test.com]`, and the page has `a.test.com` as a referrer domain. |
+| No previous campaign. | A user's initial visit. |
+| There is an introduction of new UTM parameter or `clickId` parameter. | If any utm parameters or `clickId` parameters has been dropped during a session, we will unset it. |
+| The referrer domain changes to a new one. | Referrer domain changed from `a.test.com` to `b.test-new.com`|
+
+Conversely, attribution tracking will not be conducted under any of the following conditions:
```
Comment (markzegarelli) [inline docs/data/sdks/browser-2/index.md:455]:
```suggestion
Amplitude doesn't track marketing attribution under any of the following conditions:
```

**Review by markzegarelli — APPROVED** [review]

File: docs/data/sdks/browser-2/index.md [hunk 4]
```diff
@@ -437,4 +437,29 @@ To disable offline mode, add `offline: amplitude.Types.OfflineDisabled` to the `
...
\ No newline at end of file
+```
+
+### Marketing Attribution Tracking
+
+Amplitude tracks marketing attribution by default. Once you enable marketing attribution tracking, Amplitude generates `identify` events to assign the campaign value in certain cases. This ensures that user properties update and influence future events. 
+For more information, see the scenarios outlined below that demonstrate when Amplitude does or doesn't track marketing attribution. These examples are illustrative, not exhaustive:
+
+Amplitude tracks marketing attribution by default. Once you enable marketing attribution tracking, Amplitude generates `identify` events to assign the campaign value in certain cases. This ensures that user properties update and influence future events. 
+For more information, see the scenarios outlined below that demonstrate when Amplitude does or doesn't track marketing attribution. These examples are illustrative, not exhaustive.
```
Comment (Mercy811) [inline docs/data/sdks/browser-2/index.md:448]: Duplicated?

File: docs/data/sdks/browser-2/index.md [hunk 5]
```diff
@@ -437,4 +437,29 @@ To disable offline mode, add `offline: amplitude.Types.OfflineDisabled` to the `
...
+
+Tracking occurs when either of the following applies:
+
+|Rule|Example|
+|-|-|
+| A specific referrer domain is explicitly excluded.| When config the `config.defaultTracking.attribution.excludeReferrers` to include `[a.test.com]`, and the page has `a.test.com` as a referrer domain. |
```
Comment (Mercy811) [inline docs/data/sdks/browser-2/index.md:454]: It looks like this is not a new campaign, `isNewCampaign()` will return false [here ](https://github.com/amplitude/Amplitude-TypeScript/blob/e8216bc65399106214cc5c0ae4d7e8d053029af4/packages/analytics-client-common/src/attribution/campaign-tracker.ts#L55-L57) and will return early [here](https://github.com/amplitude/Amplitude-TypeScript/blob/e8216bc65399106214cc5c0ae4d7e8d053029af4/packages/analytics-client-common/src/attribution/campaign-tracker.ts#L111-L113) before track an identify event.

**Review by Mercy811 — CHANGES_REQUESTED** [review]

File: docs/data/sdks/browser-2/index.md [hunk 6]
```diff
@@ -437,4 +437,29 @@ To disable offline mode, add `offline: amplitude.Types.OfflineDisabled` to the `
...
+| No previous campaign. | A user's initial visit. |
+| There is an introduction of new UTM parameter or `clickId` parameter. | If any utm parameters or `clickId` parameters has been dropped during a session, we will unset it. |
```
Comment (Mercy811) [inline docs/data/sdks/browser-2/index.md:456]: Remove code style to keep consistent

https://www.docs.developers.amplitude.com/data/sdks/browser-2/#click-id-parameters
```suggestion
| There is an introduction of new UTM parameter or Click ID parameter. | If any utm parameters or Click ID parameters has been dropped during a session, we will unset it. |
```

File: docs/data/sdks/browser-2/index.md (hunk 5 above)
Comment (yuhao900914) [reply docs/data/sdks/browser-2/index.md:454]: campaign-tracker.ts has been deprecated. We are not using it in browser sdk.

Comment (Mercy811) [reply docs/data/sdks/browser-2/index.md:451]: web attribution plugin should have the same logic https://github.com/amplitude/Amplitude-TypeScript/blob/e8216bc65399106214cc5c0ae4d7e8d053029af4/packages/plugin-web-attribution-browser/src/helpers.ts#L34

Comment (Mercy811) [reply docs/data/sdks/browser-2/index.md]: Sync with Alyssa and Alyssa moved this line to examples of not tracking web attribution at https://github.com/amplitude/amplitude-dev-center/pull/242/commits/ffa12b04cf81604cb570973b5182e8122a1ec9b0

**Review by Mercy811 — APPROVED** [review]
Thank you Alyssa for adding this.

**Pull request merged**